### PR TITLE
Diff format

### PR DESCRIPTION
**Pull request by abeshi-softwire**
Added a `diffTime` and `diffTimeFormatted` to `Journey`, and switch both iOS and Android code to using the shared formatted duration.

### CODE REVIEW

File: SharedCode/src/commonMain/kotlin/common.kt [hunk 1]
```diff
@@ -68,26 +68,44 @@ fun toHumanReadableDate(epochMillis: Long, dateInfo: DateFormatInfo): String {
...
 
+fun formatDiff(date1Millis: Long, date2Millis: Long): String {
+    val date1 = DateTime.fromUnix(date1Millis)
+    val date2 = DateTime.fromUnix(date2Millis)
+    val diff = date2 - date1
+    if (date1.dayOfYear == date2.dayOfYear || diff.hours < 12) {
+        if (diff.hours < 1) {
+            return "${diff.minutes.toInt()} min"
+        } else {
+            return "${diff.hours.toInt()}:${"${(diff.minutes.toInt() % 60)}".padStart(2, '0')}"
```
Comment (thevirtuoso1973) [inline SharedCode/src/commonMain/kotlin/common.kt]: I'm not sure that giving the precise difference is better than just saying 'in 2 hours'. It looks a _little_ cluttered.

**Review by thevirtuoso1973 — COMMENTED** [review]
Tested on Android and working, but see comment.

File: SharedCode/src/commonMain/kotlin/common.kt (hunk 1 above)
Comment (abeshi-softwire) [reply SharedCode/src/commonMain/kotlin/common.kt]: That's fair, I was just wondering for train duration you probably want minutes as well. Maybe I'll add a flag.

Comment (abeshi-softwire) [reply SharedCode/src/commonMain/kotlin/common.kt]: Added the flag

**Review by thevirtuoso1973 — APPROVED** [review]
'precise = true' hmm 🤔

**Comment by thevirtuoso1973** [comment]
> 'precise = true' hmm 🤔

Oh ignore this, I was looking at the wrong 'precise' assignment.

**Pull request merged**